worldguard-core/src/main/java/com/sk89q/worldguard/commands/task/RegionLister.java

```diff
@@ -144,20 +151,29 @@ public boolean isContainedWithin(DefaultDomain domain) throws CommandException {
         };
     }
 
+    public void filterIdByMatch(String idFilter) {
+        this.idFilter = idFilter;
+    }
+
     @Override
     public Integer call() throws Exception {
         Map<String, ProtectedRegion> regions = manager.getRegions();
+        Collection<ProtectedRegion> iterableRegions = manager.getRegions().values();
```

Comment:
should use `regions` again, `getRegions` is a full copy so this is both a performance and race condition.